module/VuFind/src/VuFind/Controller/UpgradeController.php

```diff
@@ -733,13 +740,16 @@ public function fixmetadataAction()
             foreach ($problems as $problem) {
                 try {
                     $driver = $this->getRecordLoader()
-                        ->load($problem->record_id, $problem->source);
-                    $problem->assignMetadata($driver, $converter)->save();
+                        ->load($problem->getRecordId(), $problem->getSource());
+                    $resourceService->assignMetadata($driver, $converter, $problem);
+                    $resourceService->persistEntity($problem);
                 } catch (RecordMissingException $e) {
                     $this->session->warnings->append(
                         "Unable to load metadata for record "
                         . "{$problem->source}:{$problem->record_id}"
                     );
+                } catch (\Exception $e) {
+                    $this->logError('Could not save resource: ' . $e->getMessage());
```

Comment:
Should you rethrow the exception after logging the error so you don't fall through into the following logic?

Reply:
You're right, that's a good idea. But I think if we do that, we should also add a warning message to the session so the end user will know that something happened. Maybe something like:

```
                    $this->session->warnings->append(
                        "Problem saving metadata updates for record "
                        . "{$problem->source}:{$problem->record_id}"
                    );
```